types/urijs/index.d.ts

```diff
@@ -229,7 +230,8 @@ interface URI {
     scheme(): string;
     scheme(protocol: string): URI;
     search(): string;
-    search(qry: string | URI.QueryDataMap | ((qryObject: URI.QueryDataMap) => (URI.QueryDataMap | undefined))): URI;
+    // tslint:disable-next-line void-return
+    search(qry: string | URI.QueryDataMap | ((qryObject: URI.QueryDataMap) => (URI.QueryDataMap | void))): URI;
```

Comment:
Ditto

Reply:
I'd disable such rule `// tslint:disable-next-line void-return` directly in the file. If it's the way the library works, we do not have to circumnavigate anything.